src/downloadmanager.cpp

```diff
@@ -440,7 +455,10 @@ bool DownloadManager::addDownload(const QStringList &URLs, QString gameName,
   h2Conf.setStreamReceiveWindowSize(16777215);
   QNetworkRequest request(preferredUrl);
   request.setHeader(QNetworkRequest::UserAgentHeader, m_NexusInterface->getAccessManager()->userAgent());
+  request.setAttribute(QNetworkRequest::CacheSaveControlAttribute, false);
```

Comment:
Any reason for these two added attributes?

Reply:
Thae same reason they are used in Nexus downloads https://github.com/ModOrganizer2/modorganizer/blob/qt6/src/nexusinterface.cpp#L821 . Spent half a day pulling my hair out on this.

By default, MO2 caches repeating requests, especially with small files. When this happens, the `finished()` signal is sent immediately, causing `downloadFinished` to fire before downloading a single byte. Then the check for size==0 kicks in here `https://github.com/ModOrganizer2/modorganizer/blob/qt6/src/downloadmanager.cpp#L2100`, and the download gets marked as failed without showing a single message, because messages don't cover this case.

These two lines disable cache for the request.